tortoisebeacon/weakcoin/handler.go

```diff
@@ -0,0 +1,67 @@
+package weakcoin
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/p2p/service"
+)
+
+// HandleSerializedMessage defines method to handle Tortoise Beacon Weak Coin Messages from gossip.
+func (wc *WeakCoin) HandleSerializedMessage(ctx context.Context, data service.GossipMessage, sync service.Fetcher) {
+	wc.logger.With().Debug("received weak coin message",
+		log.String("from", data.Sender().String()),
+		log.Binary("data", data.Bytes()),
+	)
+
+	var message Message
+	if err := types.BytesToInterface(data.Bytes(), &message); err != nil {
+		wc.logger.With().Debug("received invalid weak coin message",
+			log.Binary("message", data.Bytes()),
```

Comment:
the problem is that i don't want to encode always. otherwise someone can send any junk and waste node cpu encoding it to hex. with log.Binary cpu will be spent only if log level is Debug, which shouldn't be set to it on production node